scripts/scil_denoise_sf.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+Script to compute per-vertices hemisphere-aware (asymmetric) Gaussian
+filtering of spherical functions (SF) given an array of spherical harmonics
+(SH) coefficients.
+
+The resulting SF can be expressed using a full SH basis (to keep the
+asymmetry resulting from the filtering) or a symmetric SH basis (where the
+effect of the filtering is a simple denoising).
+"""
+
+import argparse
+import logging
+import os
+
+import nibabel as nib
+import numpy as np
+
+from scilpy.io.utils import (add_overwrite_arg, assert_inputs_exist,
+                             assert_outputs_exist, add_sh_basis_args)
+
+from scilpy.denoise.asym_enhancement import local_asym_gaussian_filtering
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(
+        description=__doc__, formatter_class=argparse.RawTextHelpFormatter)
+
+    p.add_argument('in_sh',
+                   help='Path to the input file.')
+
+    p.add_argument('--out_dir', default='.',
+                   help='Output directory. Default is current directory.')
+
+    p.add_argument('--out_sh', default='out_filtered.nii.gz',
+                   help='File name for averaged signal. [%(default)s]')
+
+    p.add_argument('--out_mask', default='mask.nii.gz',
+                   help='File name for output mask. [%(default)s]')
+
+    p.add_argument('--mask_eps', default=1e-16,
+                   help='Threshold on SH coefficients norm for output mask.'
+                   ' [%(default)s]')
+
+    p.add_argument(
+        '--sh_order', default=8, type=int,
```

Comment:
sh_order can be determined pretty easily, why ask for that?
Also we can't specify the sh_basis?

Would be nice if it supported both
tournier07 -> tournier07full
descoteaux07 -> descoteaux07full

Reply:
I made the choice to infer the nature of the basis (symmetric or full) from the number of coefficients instead of the maximum SH order, but we could replace `--sh_order` by `--full` to rather specify when the input is in full basis. That way we could determine the SH order from the number of SH coefficients. I could also investigate if it is possible to find out if the SH basis is symmetric or full __and__ the maximum order, but I'm afraid there could be collisions, i.e. where a symmetric basis of _some order_ has the same number of coefficients as a full basis of _some other order_.

Also SH basis argument is on [line 72](https://github.com/scilus/scilpy/blob/93e284ea25fb7cce46c8e4479fc8cd5a091e4ef6/scripts/scil_denoise_sf.py#L72)!

Reply:
Thanks for the response, sorry I forgat that we have a function to add the basis.
Now I understand better why its important, I didn't know there was conflict.

I have no preference between --sh_order or --full. Leave it like that until someone either confirm or request differently.

Reply:
I was just checking scil_visualise_fodf.py and there is a --full option. If these options reflect the same effect please use the same name.

Reply:
No need for `--full` option because there is a `--sh_order` argument. Full vs symmetric basis can be inferred from the SH order.